host/allwinner/detect.go

```diff
@@ -65,8 +74,20 @@ func (d *detectionS) do() {
 				if strings.Contains(c, "sun5i-r8") {
 					d.isR8 = true
 				}
+				if strings.Contains(c, "sun7i-a20") {
+					d.isA20 = true
+				}
+			}
+
+			// The kernel in the image that comes pre-installed on the pcDuino3 Nano
```

Comment:
Please add a bit more info like:

The kernel in the image that comes pre-installed on the pcDuino3 Nano is using an old 3.x kernel that doesn't expose the device-tree in procfs, so do an extra check in cpuinfo. Distros based on 4.x kernels do expose it.

Reply:
In fact, since recent kernels do the right thing, I'd prefer to do the "right" check with `sun7i-a20` first and only fallback if it's not the case.